### PR TITLE
Write template locals to json file

### PR DESCRIPTION
**Pull request by tntrex**
Hello

I faced with `RuntimeException` error when my locals object getting so big. And I'm find out that my very large `obj` argument passed directly to cli. So I added `localsJsonFile` option that write locals to json file and pass it as `obj` argument.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -262,6 +262,22 @@ magic methods will be striginfied in JSON as simple objects) and you will not be
 features like mixed indent, pre/post render hooks but in this mode you will get exact same
 output as in pug-js.
 
+### Write locals object to json file with pugjs
```
Comment (kylekatarnls) [inline README.md]: Hi, as this part comes with pugjs=>true, it should be a sub-section of the previous one, so I would mark it as a h4 (`####`)

File: src/Jade/Engine/PugJsEngine.php [hunk 2]
```diff
@@ -54,7 +54,15 @@ protected function getPugJsOptions(&$input, &$filename, &$vars, &$pug)
             'out' => $workDirectory,
         );
         if (!empty($vars)) {
-            $options['obj'] = json_encode($vars, JSON_UNESCAPED_SLASHES);
+            $locals = json_encode($vars, JSON_UNESCAPED_SLASHES);
+
+            $options['obj'] = $locals;
+
+            if ($this->options['localsJsonFile']) {
+                $optionsFile = $workDirectory . '/options-' . hash('md5', time()) . '.json';
```
Comment (kylekatarnls) [inline src/Jade/Engine/PugJsEngine.php]: Hi, I find the solution fine but `time()` should not be used for storage, imagine the following:
```php
$pug = new Pug(array(
    'pugjs' => true,
    'localsJsonFile' => true
);
$pug->render('foo.pug', $variablesForFoo);
$pug->render('bar.pug', $variablesForBar);
```
time() will probably be the same for foo and bar. I think here, tempnam would be appropriate: http://php.net/manual/en/function.tempnam.php

File: README.md [hunk 3]
```diff
@@ -262,6 +262,22 @@ magic methods will be striginfied in JSON as simple objects) and you will not be
...
+
+If your locals object is large it may cause a `RuntimeException` error. This is because
+locals object passed directly to pug-cli as argument. To fix this problem you can use
+`localsJsonFile` option:
+
+```php
+$pug = new Pug(array(
+    'expressionLanguage' => 'js',
```
Comment (kylekatarnls) [inline README.md]: expressionLanguage is automatically js and cannot be changed if pugjs => true, so this option should be omitted here.

**Review by kylekatarnls — CHANGES_REQUESTED** [review]
Hi, thanks for your help. I'm sorry you get trouble with the rendering.

If it's not too much to ask to you I would appreciate you fix the little points I give to you in review.

CodeClimate also suggest to splig the renderWithJs method that become too complex. But don't worry about that, I can merge your pull-request and see how to refactor afterward.

File: README.md (hunk 1 above)
Comment (tntrex) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (tntrex) [reply README.md]: I have removed this line

**Review by kylekatarnls — APPROVED** [review]

File: src/Jade/Engine/PugJsEngine.php (hunk 2 above)
Comment (tntrex) [reply src/Jade/Engine/PugJsEngine.php]: `hash` was replaced with `mt_rand` like in https://github.com/pug-php/pug/blob/master/src/Jade/Engine/PugJsEngine.php#L46

**Comment by kylekatarnls** [comment]
Thanks a lot, your option is available at the version 2.6.5.

**Comment by tntrex** [comment]
still 2.6.4 at packagist :(

**Comment by kylekatarnls** [comment]
It's usually very fast to update, I put a manual update.

**Pull request merged**